odoo/fields.py

```diff
@@ -2223,7 +2223,7 @@ def _related_domain(self):
     _description_context = property(attrgetter('context'))
 
     def _description_domain(self, env):
-        if self.check_company and not self.domain:
+        if self.check_company and not self.domain and not self.comodel_name == "res.users":
```

Comment:
i agree the default domain is actually more painful than anything
we could also use "[('company_ids', 'in', company_id)]"

Reply:
What about the case when company_id is not set? You wouldn't be able to select any user no?

Reply:
yeah
"['|', ('company_ids', 'in', company_id), ('company_id', '=', False)]"

Reply:
Nope, because the `'company_id'` here is the company of the user, which is set, not the company of the current record ;).

Reply:
..right

Reply:
The only way to solve this would have a domain field which can be computed depending on the record itself (which is only feasible through onchanges currently).  The domain attribute supports callable, but the callable receives the environment, not the record itself.

Reply:
Therefore, we can only disable this for `res.users` currently and have the check_company refuse bad values :disappointed:.

Reply:
we could add "[('company_ids', 'in', company_id)]" if company_id is required on the main record and disable the domain if not

Reply:
How can we know in a field definition if another field is defined as required :thinking: ?

Reply:
so let's just disable the default domain for res.users

Reply:
I think I managed to make it work, and it's not as ugly as I thought.